designs/move-cluster-specific-code-out-of-manager.md

```diff
@@ -0,0 +1,209 @@
+Move cluster-specific code out of the manager
+===================
+
+Currently, the `./pkg/manager.Manager` has two purposes:
+
+* Handle running controllers/other runnables and managing their lifecycle
+* Setting up various things to interact with the Kubernetes cluster,
+  for example a `Client` and a `Cache`
+
+This works very well when building controllers that talk to a single cluster,
+however some use-cases require controllers that interact with more than
+one cluster. This multi-cluster usecase is very awkward today, because it
+requires to construct one manager per cluster and adding all subsequent
+managers to the first one.
+
+This document proposes to move all cluster-specific code out of the manager
+and into a new package and interface, that then gets embedded into the manager.
+This allows to keep the usage for single-cluster cases the same and introduce
+this change in a backwards-compatible manner.
+
+Furthermore, the manager gets extended to start all caches before any other
+`runnables` are started.
+
+
+The new `ClusterConnector` interface will look like this:
```

Comment:
Ideas for a better name are very welcome :)

Reply:
Or maybe`PeerCluster` or `ClusterPeer`? :)

Reply:
Regarding naming: What about just `Cluster`? Its pretty much what this represents

Reply:
Cluster it is now